### ISSUE TITLE
pdk executable not installed in path on Debian (8.8 Jessie)

### ISSUE THREAD

**Issue by bittner** [puppetlabs/pdk#272]
I'm using pdk v1.0.1 on Debian 8.8 Jessie, installed via the official `pdk_1.0.1.0-1trusty_amd64.deb` package.

When I install the toolkit via `sudo dpk -i pdk_1.0.1.0-1trusty_amd64.deb`
Then the installation succeeds
But there is no executable `pdk` found in my path in the shell.

I have to add a symlink to `/usr/local/bin` to make this work:
``` bash
sudo ln -sv /opt/puppetlabs/bin/pdk /usr/local/bin/
```

**Comment by DavidS** [comment]
Hi,

thanks for reporting that. I can repro that on my local machine (Debian testing). The PATH is configured in `/etc/profile.d/pdk.sh` which is only evaluated on login shells, but not in terminals under X.

From reading the [Debian Policy](https://www.debian.org/doc/debian-policy/index.html#environment-variables) and the [FHS](https://www.debian.org/doc/packaging-manuals/fhs/fhs-2.3.html#OPTADDONAPPLICATIONSOFTWAREPACKAGES) your fix seems to be the proper response.

While researching further options, I stumbled upon [this section](https://www.debian.org/doc/manuals/debian-reference/ch07.en.html#_customizing_the_x_session_new_method) in the Debian Reference manual which claims that gdm3 would evaluate `/etc/profile` as part of the start up process.

Since that also doesn't help everyone, we could drop a copy of  `/etc/profile.d/pdk.sh` in `/etc/X11/Xsession.d`. It is already idempotent, and would not change the `PATH` if `/etc/profile` were evaluated in the same session.

What do you think?

**Comment by DavidS** [comment]
I've added the file to my Xsession, and after a new login, the pdk is available on my PATH:

```
david@davids:~$ cat /etc/X11/Xsession.d/20pdk 
# Add /opt/puppetlabs/bin to the path for sh compatible users

if ! echo "$PATH" | grep -q /opt/puppetlabs/bin ; then
  export PATH=$PATH:/opt/puppetlabs/bin
fi
david@davids:~$ which pdk
/opt/puppetlabs/bin/pdk
david@davids:~$ 
```

**Comment by bittner** [comment]
Whatever makes it work as a tool is fine with me. I  just wanted to make sure you get feedback, and we make the toolkit helpful and work for everyone. Thanks! :+1:

**Comment by rodjek** [comment]
:+1: to adding to Xsession.d over symlinking into `/usr/local/bin/`

**Comment by DavidS** [comment]
This is being tracked in https://tickets.puppetlabs.com/browse/PDK-446

**Comment by bittner** [comment]
Works on Ubuntu 17.04 Zesty out-of-the-box. No symlinking required.

**Comment by bittner** [comment]
Somewhat related, the other included tools (the ones listed in the table on top of the README) are obviously meant to be run via `pdk` (e.g. `puppet-lint`, `rubocop`, ...), because they are all located in `/opt/puppetlabs/pdk/share/cache/ruby/2.1.0/bin/`, not in `/opt/puppetlabs/bin/`.

The problem with that is that `puppet-lint` normally also validates a control repository, not only a module. We simply run `puppet-lint .` in the repository root, usually. Now, `pdk validate` does not like this idea / feature:

```
$ pdk validate .
pdk (FATAL): This command must be run from inside a module (no metadata.json found)
```
One solution would be to make `puppet-lint` directly accessible by placing it in `/opt/puppetlabs/bin/`. Not sure how you guys want us to validate control repositories.

**Comment by DavidS** [comment]
PDK-446 has been resolved. The next package drop will contain the fixes.

**Comment by bittner** [comment]
For the included tools rant I've opened issue #289.
